### DISCUSSION TITLE
How to share sever-side-only codes between getServerSideProps?

### DISCUSSION THREAD

**Discussion by youminkim** [vercel/next.js#22332] [Help]
Hi, a novice here and this could be an odd question. 

I have repeatedly used server-side code like this function: 
```
async function accountInfo(phone) {
    const { Datastore } = require('@google-cloud/datastore');
    const datastore = new Datastore();
    const [account] = await datastore.get(
        datastore.key(['Account', Number(phone)])
    )
    return account
}
```

I usually use this function in my `getServerSideProps`
```
export async function getServerSideProps(context) {
    ...
    const account = await accountInfo(phone)
    ...
}
```

I have several other pages using  `getServerSideProps` and I wanted to share `accountInfo` function with `export`. However, there are side effects:
1. Once I put `export` to the function, the code is visible in client side. I verified with https://next-code-elimination.vercel.app/
2. I got this error when I import the function in other pages:
```
error - ./node_modules/google-auth-library/build/src/auth/googleauth.js:17:0
Module not found: Can't resolve 'child_process'
```

I believe that there should be a way to share such server-side codes between files. Can you help?

**Comment by stigkj** [comment]
As long as you only use it in `getServerSideProps` it should just work. Can you post an example that shows what happens for you?
